### PR TITLE
[REVIEW] multigraph_changes

### LINKED ISSUE #589: [FEA] Add a (Di)Graph constructor that accepts Multi(Di)Graph and one edge_attr
Users may want to call an existing single-dimension algorithm on a graph with multiple edge attributes (`Multi(Di)Graph`). That raises the question of which edge attribute should be used in this case. 

We should have a `(Di)Graph` constructor that accepts : 
- a `Multi(Di)Graph`
- valid `src` and `dst` columns within this `Multi(Di)Graph`. 
- one optional user-specified `edge_attribute` within this `Multi(Di)Graph`. 

We should avoid copying of the GPU DataFrame though. The `(Di)Graph` would just point to the relevant `Multi(Di)Graph` data.

The user can then pass this `(Di)Graph` to any existing analytics.

### PR DESCRIPTION
**Pull request by Iroy30**
This PR enables user to create (Di)Graph from Multi(Di)Graph with an option of providing any one column of the multigraph as the edge_attr.
Closes #589

### CODE REVIEW

**Comment by afender** [comment]
Please add relevant flags in the PR title

**Comment by Iroy30** [comment]
rerun tests

File: python/cugraph/structure/graph.py [hunk 1]
```diff
@@ -72,6 +76,14 @@ def __init__(self, symmetrized=False, bipartite=False, multi=False,
         self.edgelist = None
         self.adjlist = None
         self.transposedadjlist = None
+        if m_graph is not None:
+            if edge_attr is None:
+                raise Exception('edge_attr not provided')
```
Comment (afender) [inline python/cugraph/structure/graph.py]: It should be ok to have graphs without edge attributes / weights. Several algos don't use weights at all.

File: python/cugraph/structure/graph.py [hunk 2]
```diff
@@ -113,7 +125,7 @@ def from_cudf_edgelist(self, input_df, source='source',
         source : str
             source argument is source column name
         destination : str
-            destination argument is destination column name.
+            target argument is destination column name.
```
Comment (afender) [inline python/cugraph/structure/graph.py]: We should use `destination` consistently

File: python/cugraph/structure/graph.py [hunk 3]
```diff
@@ -158,7 +176,7 @@ def add_edge_list(self, source, destination, value=None):
  Use from_cudf_edgelist instead')
         input_df = cudf.DataFrame()
         input_df['source'] = source
-        input_df['destination'] = destination
+        input_df['target'] = destination
```
Comment (afender) [inline python/cugraph/structure/graph.py]: destination

File: python/cugraph/structure/graph.py (hunk 1 above)
Comment (Iroy30) [reply python/cugraph/structure/graph.py]: fixed with better exception throwing and removing edge_attr constraint for graph constructor with multigraph

**Comment by Iroy30** [comment]
rerun tests

**Comment by Iroy30** [comment]
rerun tests

**Comment by Iroy30** [comment]
rerun tests

File: python/cugraph/structure/graph.py (hunk 2 above)
Comment (Iroy30) [reply python/cugraph/structure/graph.py]: fixed

File: python/cugraph/structure/graph.py (hunk 3 above)
Comment (Iroy30) [reply python/cugraph/structure/graph.py]: fixed

File: python/cugraph/structure/graph.py (hunk 1 above)
Comment (Iroy30) [reply python/cugraph/structure/graph.py]: Added test to test the feature

**Pull request merged**